### PR TITLE
CONN-2442 import wizard improvement

### PR DESCRIPTION
**Pull request by trantang**
rename referenceNumber to CN
merge CSR
fix import-issue
tmp folder for import wizard

### CODE REVIEW

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 1]
```diff
@@ -630,13 +643,21 @@ public SimpleCertificateResponseMessageType importToKeystore(ImportToKeystoreReq
...
+
         try {
             Certificate publicKey = CertificateUtil.createCertificate(request.getServerCert());
-            PrivateKey privateKey = CertificateUtil.getPrivateKeyEntry(getTempKeystore(), alias,
-                System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY)).getPrivateKey();
-
-            if(null == publicKey || null == privateKey){
-                return buildSimpleResponse(false, "Error certificate does not have keypair: "+alias);
+            if (null == publicKey) {
+                return buildSimpleResponse(false, "Error server certificate is not upload.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: "is not upload" -> "was not uploaded"
Remove the word "Error" to keep consistent with other messages.

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 2]
```diff
@@ -630,13 +643,21 @@ public SimpleCertificateResponseMessageType importToKeystore(ImportToKeystoreReq
...
         }
+
+        PrivateKey privateKey = readPrivateKey(alias, getPasswordKeystore());
+        if (null == privateKey) {
+            return buildSimpleResponse(false, "Error no privatekey recorded for alias: " + alias);
+        }
+
+        tempKeystore = null;
+        if (!copyFile(FILE_JKS_GATEWAY, FILE_JKS_GATEWAY_NEW)) {
+            return buildSimpleResponse(false, "unable to make a copy of gateway-jks.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: "Unable to copy keystore to temporary file" - the user might not have their file called "gateway", so lets generalize this error.

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 3]
```diff
@@ -599,18 +613,17 @@ private X509Certificate generateCertificate(KeyPair keypair, String dn, int days
...
-        }
-        if (deleteCaCert.exists() && deleteCaCert.delete()) {
-            deleted = true;
+        if (tmpFolder.exists()) {
+            try {
+                FileUtils.deleteDirectory(tmpFolder);
+                deleted = true;
+            } catch (IOException e) {
+                LOG.error("Error while cleaning up the tmp folder: {}", e.getLocalizedMessage(), e);
+                return buildSimpleResponse(false, "Error whild cleaning up the temporary folder.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: whild -> "while"

Can't comment directly due to unchanged line - 
"Temporary files are deleted: successful." : "Temporary files does not exist." ->
"Temporary files were deleted successfully." : "Temporary files did not exist. Skipping temporary file cleanup"

The latter message makes it sound as if this is an error.

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 4]
```diff
@@ -630,13 +643,21 @@ public SimpleCertificateResponseMessageType importToKeystore(ImportToKeystoreReq
         if (StringUtils.isBlank(alias) || null == request.getServerCert()) {
             return buildSimpleResponse(false, "Alias and Server Certificate are required.");
         }
+
+        PrivateKey privateKey = readPrivateKey(alias, getPasswordKeystore());
+        if (null == privateKey) {
+            return buildSimpleResponse(false, "Error no privatekey recorded for alias: " + alias);
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: Remove the word "error" to keep consistent with other messages.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/certificateManager.xhtml [hunk 5]
```diff
@@ -348,7 +348,7 @@
                                             <p:autoUpdate />
                                         </p:messages>
                                         <div class="content">
-                                            <h:outputText>Creating a certificate creates a temporary certificate. This temporary certificate will be used to replace with the existing keystore file. Please skip to Create CSR if existing certificate will be used.</h:outputText>
+                                            <h:outputText>This creates a temporary certificate that will be used to replace existing keystore file. Please skip to Create CSR if existing certificate will be used.</h:outputText>
```
Comment (plobre) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/certificateManager.xhtml]: "Please skip to Create CSR if existing certificate will be used." ->"Please skip to Create CSR if an existing certificate will be used."

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 6]
```diff
@@ -711,8 +730,7 @@ public SimpleCertificateResponseMessageType importToTruststore(ImportToTruststor
                 getTempTruststore().setCertificateEntry(entryCert.getKey(), entryCert.getValue());
             }
 
-            CoreHelpUtils.saveJksTo(getTempTruststore(), System.getProperty(CertificateManagerImpl.TRUST_STORE_PASSWORD_KEY),
-                FILE_JKS_CACERTS_NEW);
+            CoreHelpUtils.saveJksTo(getTempTruststore(), getPasswordTruststore(), FILE_JKS_CACERTS_NEW);
         }catch(CertificateManagerException | KeyStoreException | UtilException | CertificateEncodingException e){
             LOG.error("error while importing to the temporary truststore: {}",e.getLocalizedMessage(), e);
             return buildSimpleResponse(false, "error while importing to the temporary truststore.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: error -> "Error occurred"

Same applies for line 717.

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 7]
```diff
@@ -683,6 +702,7 @@ public SimpleCertificateResponseMessageType importToTruststore(ImportToTruststor
             || null == request.getRootCert()) {
             return buildSimpleResponse(false, "Alias, CA Root and CA Intermediate are required.");
         }
+        tempTruststore = null;
         if(!copyFile(FILE_JKS_CACERTS, FILE_JKS_CACERTS_NEW)){
             return buildSimpleResponse(false, "unable to create the temporary Truststore.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: unable -> "Unable"

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 8]
```diff
@@ -721,4 +739,49 @@ public SimpleCertificateResponseMessageType importToTruststore(ImportToTruststor
         return buildSimpleResponse(true, "Import is successful into the temporary truststore.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: "Import is successful into the temporary truststore." -> "Import into the temporary truststore is successful."

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 9]
```diff
@@ -658,11 +679,9 @@ public SimpleCertificateResponseMessageType importToKeystore(ImportToKeystoreReq
...
-            getTempKeystore().setKeyEntry(alias, privateKey,
-                System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY).toCharArray(),
+            getTempKeystore().setKeyEntry(alias, privateKey, getPasswordKeystore().toCharArray(),
                 CoreHelpUtils.getCertificateChain(publicKey));
-            CoreHelpUtils.saveJksTo(getTempKeystore(),
-                System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY), FILE_JKS_GATEWAY_NEW);
+            CoreHelpUtils.saveJksTo(getTempKeystore(), getPasswordKeystore(), FILE_JKS_GATEWAY_NEW);
         } catch (CertificateManagerException | UtilException | KeyStoreException | CertificateEncodingException e) {
             LOG.error("Error during import to Keystore: {}",e.getLocalizedMessage(), e);
             return buildSimpleResponse(false, "Error during import to Keystore.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: "Error during import to Keystore." -> "Error occurred importing to keystore"

**Review by plobre — CHANGES_REQUESTED** [review]
Needs a few spelling / grammar corrections, but after that should be good to go.

**Review by ptambellini01 — APPROVED** [review]

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 10]
```diff
@@ -630,13 +644,21 @@ public SimpleCertificateResponseMessageType importToKeystore(ImportToKeystoreReq
...
+
         try {
             Certificate publicKey = CertificateUtil.createCertificate(request.getServerCert());
-            PrivateKey privateKey = CertificateUtil.getPrivateKeyEntry(getTempKeystore(), alias,
-                System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY)).getPrivateKey();
-
-            if(null == publicKey || null == privateKey){
-                return buildSimpleResponse(false, "Error certificate does not have keypair: "+alias);
+            if (null == publicKey) {
+                return buildSimpleResponse(false, "Server certificate was not upload.");
```
Comment (TabassumJafri) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: Server certificate was not uploaded

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 11]
```diff
@@ -658,14 +680,12 @@ public SimpleCertificateResponseMessageType importToKeystore(ImportToKeystoreReq
...
+            getTempKeystore().setKeyEntry(alias, privateKey, getPasswordKeystore().toCharArray(),
                 CoreHelpUtils.getCertificateChain(publicKey));
-            CoreHelpUtils.saveJksTo(getTempKeystore(),
-                System.getProperty(CertificateManagerImpl.KEY_STORE_PASSWORD_KEY), FILE_JKS_GATEWAY_NEW);
+            CoreHelpUtils.saveJksTo(getTempKeystore(), getPasswordKeystore(), FILE_JKS_GATEWAY_NEW);
         } catch (CertificateManagerException | UtilException | KeyStoreException | CertificateEncodingException e) {
-            LOG.error("Error during import to Keystore: {}",e.getLocalizedMessage(), e);
-            return buildSimpleResponse(false, "Error during import to Keystore.");
+            LOG.error("Error occurred importing to keystore: {}", e.getLocalizedMessage(), e);
+            return buildSimpleResponse(false, "Error occurred importing to keystore.");
```
Comment (TabassumJafri) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: Error occurred while importing to keystore

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

**Review by plobre — APPROVED** [review]

**Review by TabassumJafri — APPROVED** [review]

**Pull request merged**